packs/vector/templates/vector.nomad.tpl

```diff
@@ -0,0 +1,140 @@
+job [[ template "full_job_name" . ]] {
+
+  region      = [[ .vector.region | quote ]]
+  datacenters = [ [[ range $idx, $dc := .vector.datacenters ]][[if $idx]],[[end]][[ $dc | quote ]][[ end ]] ]
+  type        = "system"
+  namespace   = [[ .vector.namespace | quote ]]
+  [[ if .vector.constraints ]][[ range $idx, $constraint := .vector.constraints ]]
```

Comment:
It would be really great if any mandatory constraints (e.g. `docker.volumes.enabled`, `attr.kernel.name="linux"`) were here, and then user-defined constraints were additive. Unless I misunderstand, and those are not mandatory. I won't block on this alone, since there is an escape hatch, but I will make the suggestion. :)

Reply:
Hi there! I'll add the constraints, as it's necessary to mount some paths as read-only to read node metrics and, as is, it's only working on Linux.

I'll retest the pack again on my side as well, but could you please detail what is failing for you / which issue you're facing?

Thank you (:

Reply:
I found it.  This pack assumes Loki/Prometheus are running already. Or so it seems from the alloc stderr shown below. I'll retest once I have those packs deployed but in the meantime, I am thinking at the very least you might want to update your "Requirements" section in your README. You did a great job of listing all the Nomad requirements, but if these other apps have to be deployed first, some guidance in that section would be awesome. 

In the meantime, we're thinking about ways to detect these dependencies in some sort of a pre-flight check, so that we can notify the user from `nomad-pack` itself. 

```shell
Nov 08 15:51:23.838  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info"
Nov 08 15:51:23.838  INFO vector::app: Loading configs. paths=["local/config/vector.toml"]
Nov 08 15:51:23.851  INFO vector::sources::docker_logs: Capturing logs from now on. now=2021-11-08T15:51:23.851638571+00:00
Nov 08 15:51:23.851  INFO vector::sources::docker_logs: Listening to docker log events.
Nov 08 15:51:23.851  INFO vector::sources::host_metrics: PROCFS_ROOT is set in envvars. Using custom for procfs. custom="/host/proc"
Nov 08 15:51:23.851  INFO vector::sources::host_metrics: SYSFS_ROOT is set in envvars. Using custom for sysfs. custom="/host/sys"
Nov 08 15:51:23.899  INFO vector::topology::running: Running healthchecks.
Nov 08 15:51:23.900 ERROR vector::topology::builder: msg="Healthcheck: Failed Reason." error=Failed to make HTTP(S) request: error trying to connect: tcp connect error: Connection refused (os error 111) component_kind="sink" component_type="loki" component_id=loki component_name=loki
Nov 08 15:51:23.900 ERROR vector::topology::running: Sinks unhealthy.
Nov 08 15:51:41.681  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info"
Nov 08 15:51:41.681  INFO vector::app: Loading configs. paths=["local/config/vector.toml"]
Nov 08 15:51:41.691  INFO vector::sources::docker_logs: Capturing logs from now on. now=2021-11-08T15:51:41.691269528+00:00
Nov 08 15:51:41.691  INFO vector::sources::docker_logs: Listening to docker log events.
Nov 08 15:51:41.691  INFO vector::sources::host_metrics: PROCFS_ROOT is set in envvars. Using custom for procfs. custom="/host/proc"
Nov 08 15:51:41.691  INFO vector::sources::host_metrics: SYSFS_ROOT is set in envvars. Using custom for sysfs. custom="/host/sys"
Nov 08 15:51:41.749  INFO vector::topology::running: Running healthchecks.
Nov 08 15:51:41.749  INFO vector::topology::builder: Healthcheck: Disabled.
Nov 08 15:51:41.749 ERROR vector::topology::builder: msg="Healthcheck: Failed Reason." error=Failed to make HTTP(S) request: error trying to connect: tcp connect error: Connection refused (os error 111) component_kind="sink" component_type="loki" component_id=loki component_name=loki
Nov 08 15:51:41.749 ERROR vector::topology::running: Sinks unhealthy.
Nov 08 15:51:57.537  INFO vector::app: Log level is enabled. level="vector=info,codec=info,vrl=info,file_source=info,tower_limit=trace,rdkafka=info"
Nov 08 15:51:57.538  INFO vector::app: Loading configs. paths=["local/config/vector.toml"]
Nov 08 15:51:57.551  INFO vector::sources::docker_logs: Capturing logs from now on. now=2021-11-08T15:51:57.551191694+00:00
Nov 08 15:51:57.551  INFO vector::sources::docker_logs: Listening to docker log events.
Nov 08 15:51:57.551  INFO vector::sources::host_metrics: PROCFS_ROOT is set in envvars. Using custom for procfs. custom="/host/proc"
Nov 08 15:51:57.551  INFO vector::sources::host_metrics: SYSFS_ROOT is set in envvars. Using custom for sysfs. custom="/host/sys"
Nov 08 15:51:57.627  INFO vector::topology::running: Running healthchecks.
Nov 08 15:51:57.627  INFO vector::topology::builder: Healthcheck: Disabled.
Nov 08 15:51:57.630 ERROR vector::topology::builder: msg="Healthcheck: Failed Reason." error=Failed to make HTTP(S) request: error trying to connect: tcp connect error: Connection refused (os error 111) component_kind="sink" component_type="loki" component_id=loki component_name=loki
Nov 08 15:51:57.630 ERROR vector::topology::running: Sinks unhealthy.
```

Reply:
You're right, sorry about that! I should have made it clear in the README. This pack does not deploy Loki/Prometheus since there were already packs in the main branch for these two services.

I've made the 'Requirements' section clearer about having running instances of Loki and Prometheus, proposing two ways: using the existing Loki and Prometheus packs or Grafana Cloud (the solution that I'm currently using with this job).

The Docker volumes constraint has also been added to the default constraints stanza.

Thanks